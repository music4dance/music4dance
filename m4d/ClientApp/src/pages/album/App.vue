<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import SongTable from "@/components/SongTable.vue";
import AdminFooter from "@/components/AdminFooter.vue";
import { AlbumModel } from "@/models/AlbumModel";
import { TypedJSON } from "typedjson";
import { useSongSelector } from "@/composables/useSongSelector";
import { computed } from "vue";

declare const model_: string;

const model = TypedJSON.parse(model_, AlbumModel)!;

const { songs: selected, select: selectSong } = useSongSelector();

const artistRef = computed(() => `/song/artist?name=${encodeURIComponent(model.artist || "")}`);
const hidden = computed(() => (model.artist ? ["Artist"] : []));
</script>

<template>
  <PageFrame id="app">
    <h1>
      Album: {{ model.title }}
      <span v-if="model.artist"
        >by <a :href="artistRef">{{ model.artist }}</a></span
      >
    </h1>
    <SongTable
      :histories="model.histories!"
      :filter="model.filter"
      :hide-sort="true"
      :hidden-columns="hidden"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
