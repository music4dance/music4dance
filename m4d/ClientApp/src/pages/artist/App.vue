<script setup lang="ts">
import { ArtistModel } from "@/models/ArtistModel";
import { SongFilter } from "@/models/SongFilter";
import { safeDanceDatabase } from "@/helpers/DanceEnvironmentManager";
import { TypedJSON } from "typedjson";
import { Song } from "@/models/Song";
import { useSongSelector } from "@/composables/useSongSelector";

declare const model_: string;

const model = TypedJSON.parse(model_, ArtistModel)!;

const danceDB = safeDanceDatabase();
const { songs: selected, select: selectSong } = useSongSelector();

const title = `Artist: ${model.artist}`;
const filter = new SongFilter();
const histories = model.histories ?? [];
const songs = histories.map((h) => Song.fromHistory(h));
const danceIds = [
  ...songs.reduce((acc, h) => {
    const ratings = h.danceRatings;
    if (ratings) {
      ratings.forEach((r) => acc.add(r.danceId));
    }
    return acc;
  }, new Set<string>()),
];
const dances = danceIds.map((id) => danceDB.dances.find((d) => d.id === id)!);
</script>

<template>
  <PageFrame id="app" :title="title">
    <p>
      {{ histories.length }} {{ model.artist }} songs with suggestions for dances to dance to them:
      <span v-for="(dance, idx) in dances" :key="dance.id"
        ><span v-if="idx + 1 === dances.length && dances.length > 1"> and </span
        ><a :href="'/dances/' + dance.seoName">{{ dance.name }}</a
        ><span v-if="idx + 1 < dances.length && dances.length > 2">, </span></span
      >
    </p>
    <SongTable
      :histories="histories"
      :filter="filter"
      :hide-sort="true"
      :hidden-columns="['artist', 'length', 'track']"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
