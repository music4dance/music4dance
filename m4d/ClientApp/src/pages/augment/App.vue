<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import { AugmentModel } from "@/models/AugmentModel";
import { Song } from "@/models/Song";
import { SongDetailsModel } from "@/models/SongDetailsModel";
import { SongFilter } from "@/models/SongFilter";
import { SongHistory } from "@/models/SongHistory";
import SongCore from "@/pages/song/components/SongCore.vue";
import { TypedJSON } from "typedjson";
import AugmentInfo from "./components/AugmentInfo.vue";
import AugmentLookup from "./components/AugmentLookup.vue";
import AugmentSearch from "./components/AugmentSearch.vue";
import { getMenuContext } from "@/helpers/GetMenuContext";

import { computed, ref } from "vue";

enum AugmentPhase {
  lookup = "lookup",
  results = "results",
  edit = "edit",
}

declare const model_: AugmentModel;

const context = getMenuContext();

const model = TypedJSON.parse(model_, AugmentModel) ?? {};
const phase = ref<AugmentPhase>(AugmentPhase.lookup);
const songModel = ref<SongDetailsModel | null>(null);
const lastSong = ref<Song | null>(null);
const created = ref(false);
const propertiesString = ref("");
const tabIndex = ref(model.id ? 1 : 0);

const canAugment = computed(() => !!context.userName);

const editSong = (model: SongDetailsModel): void => {
  songModel.value = new SongDetailsModel({
    created: !!model.created,
    songHistory: model.songHistory,
    filter: new SongFilter(),
    userName: context.userName,
  });
  phase.value = AugmentPhase.edit;
};

const adminCreate = async (): Promise<void> => {
  songModel.value = new SongDetailsModel({
    created: true,
    songHistory: SongHistory.fromString(propertiesString.value),
    filter: new SongFilter(),
    userName: context.userName,
  });
  propertiesString.value = "";
  phase.value = AugmentPhase.edit;
};

const reset = (saved: boolean): void => {
  if (saved) {
    lastSong.value = Song.fromHistory((songModel.value! as SongDetailsModel).songHistory);
    created.value = !!songModel.value!.created;
  }
  phase.value = AugmentPhase.lookup;
  model.id = "";
  songModel.value = null;
};
</script>

<template>
  <PageFrame id="app">
    <BRow v-if="phase === 'lookup'">
      <BCol>
        <BAlert v-if="lastSong" dismissible show="10">
          Thank you for {{ created ? "adding" : "editing" }} <i>{{ lastSong.title }}</i> by
          {{ lastSong.artist }}
        </BAlert>
        <h1>Add Song</h1>
        <BTabs v-if="canAugment" v-model="tabIndex" card>
          <BTab title="by Title"
            ><BCardText><AugmentSearch @edit-song="editSong($event)"></AugmentSearch></BCardText
          ></BTab>
          <BTab title="by Id">
            <BCardText
              ><AugmentLookup :id="model.id!" @edit-song="editSong($event)"></AugmentLookup
            ></BCardText>
          </BTab>
          <BTab v-if="context.isAdmin" title="Admin">
            <BCardText>
              <p>Paste in text version of property list</p>
              <BInputGroup prepend="Properties">
                <BFormInput
                  id="admin-properties"
                  v-model="propertiesString"
                  palceholder="Song Properties TSV"
                  trim
                ></BFormInput>
                <BButton variant="primary" @click="adminCreate">Create</BButton>
              </BInputGroup>
            </BCardText>
          </BTab>
        </BTabs>
        <AugmentInfo v-else :id="model.id!"> </AugmentInfo>
      </BCol>
    </BRow>
    <div v-else>
      <BAlert show variant="success">
        <div v-if="songModel && songModel.created">
          <b>Create Song:</b> This song is new to music4dance, please fill in missing fields and
          click <b>Add Song</b> to add this song to the catalog. Remember, you must vote on at least
          one dance to add this song.
        </div>
        <div v-else>
          <b>Edit Song:</b> We found this song in the music4dance catalog, please vote on dance
          styles and add tags to improve the catalog.
        </div>
      </BAlert>
      <SongCore
        :model="songModel as SongDetailsModel"
        :start-editing="true"
        :creating="!!songModel && (songModel as SongDetailsModel).created"
        @song-saved="reset(true)"
        @cancel-changes="reset(false)"
      ></SongCore>
    </div>
    <BRow
      ><BCol>
        <BAlert show variant="warning">
          Adding songs is a new feature so if you run into bugs or have suggestions for improving
          this feature please don't hesitate to send email to
          <a href="mailto:info@music4dance.net">info@music4dance.net</a> or fill out our
          <a href="https://music4dance.blog/feedback/">feedback form</a> and we'll be happy to take
          your input as we contineu to improve the feature.
        </BAlert>
      </BCol>
    </BRow>
  </PageFrame>
</template>

<style lang="scss" scoped>
#service-id {
  width: 25rem;
}
</style>
