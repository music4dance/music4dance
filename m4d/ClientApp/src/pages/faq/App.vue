<script setup lang="ts">
import { type BreadCrumbItem, infoTrail } from "@/models/BreadCrumbItem";

const breadcrumbs: BreadCrumbItem[] = [...infoTrail, { text: "FAQ", active: true }];
</script>

<template>
  <PageFrame id="app" title="Frequently Asked Questions" :breadcrumbs="breadcrumbs">
    <div role="tablist">
      <BCard no-body class="mb-1">
        <BCardHeader header-tag="header" class="p-1" role="tab">
          <BButton v-b-toggle.accordion-1 block variant="primary">
            Where can I find songs that work with the dance style I'm currently practicing?
          </BButton>
        </BCardHeader>
        <BCollapse id="accordion-1" visible accordion="my-accordion" role="tabpanel">
          <BCardBody>
            <BCardText>
              <p>
                From the "Music Menu" at the top of the page, choose
                <a href="/song">Song Library</a>, then click on the "All Dances" button in the upper
                left to choose the style of dance you are interestaed in. Or from the same menu
                choose
                <a href="/song/advancedsearchform">Advanced Search</a>
                and include the style of dance you are interested then click search. In either case,
                the result should be a list of songs that can be danced to that style.
              </p>
              <p>
                If you would like more information check out our related blog posts.
                <a
                  href="https://music4dance.blog/question-1-im-learning-to-cha-cha-where-is-some-great-music-for-practicing/"
                >
                  I’m learning to Cha Cha, where is some great music for practicing?</a
                >
                and
                <a
                  href="https://music4dance.blog/i-am-learning-the-foxtrot-where-can-i-find-some-music/"
                >
                  I am learning the Foxtrot, where can I find some music?</a
                >
              </p>
              <p>
                The
                <a href="https://music4dance.blog/music4dance-help/advanced-search/"
                  >Advanced Search Documentation</a
                >
                describes the different search attributes in more detail.
              </p>
            </BCardText>
          </BCardBody>
        </BCollapse>
      </BCard>

      <BCard no-body class="mb-1">
        <BCardHeader header-tag="header" class="p-1" role="tab">
          <BButton v-b-toggle.accordion-2 block variant="primary">
            What dance styles work with the song I'm listening to or a specific song that I know of?
          </BButton>
        </BCardHeader>
        <BCollapse id="accordion-2" accordion="my-accordion" role="tabpanel">
          <BCardBody>
            <BCardText>
              <p>
                With the <a href="/home/counter">tempo counter tool</a> you can tap out the beat
                while listening to a song, and we'll show you a list of styles that can be danced to
                the beat you just tapped out as well as links to the dance styles' information
                pages.
              </p>
              <p>
                Or if you have a particular song in mind,
                <a href="/song/advancedsearchform">search</a> for it and if it is in the database
                look at what styles and tags are associated with it. Better yet, if you're listening
                to it in Spotify or Apple Music, you can drag the song to the text control on the
                <a href="/song">song library page</a> or
                <a href="/song/advancedsearchform">advanced search page</a> and if it's in the
                music4dance catalog, we'll take you right to it. See
                <a
                  href="https://music4dance.blog/new-feature-searching-for-a-song-from-spotify-or-itunes/"
                  >Searching for a song from Spotify or iTunes</a
                >
                for more details
              </p>
              <p>
                If you would like more information please read this blog blost
                <a
                  href="https://music4dance.blog/question-2-what-dance-styles-can-i-dance-to-my-favorite-songs/"
                  >What dance styles can I dance to my favorite song(s)?</a
                >
                and the
                <a href="https://music4dance.blog/music4dance-help/tempo-counter/">documentation</a>
                for the Tempo Counter.
              </p>
            </BCardText>
          </BCardBody>
        </BCollapse>
      </BCard>

      <BCard no-body class="mb-1">
        <BCardHeader header-tag="header" class="p-1" role="tab">
          <BButton v-b-toggle.accordion-3 block variant="primary"
            >Why should I sign up with Music4Dance?</BButton
          >
        </BCardHeader>
        <BCollapse id="accordion-3" accordion="my-accordion" role="tabpanel">
          <BCardBody>
            <BCardText>
              <p>
                <b>It's Free!</b> You'll be able to edit tags on songs and Music4Dance will remember
                the songs you like, edited, or even disliked. You'll alo be able to
                <a href="https://music4dance.blog/music4dance-help/tag-editing/">tag songs</a>,
                <a href="https://music4dance.blog/music4dance-help/dance-tags/"
                  >vote on the dancability of songs</a
                >
                and
                <a href="https://music4dance.blog/music4dance-help/add-songs/">add your own songs</a
                >. <a href="/identity/account/register">Register today!</a> We are always working on
                new features for our site and special features for our registered users.
              </p>
              <p>
                Once you've signed up, please consider
                <a href="https://music4dance.blog/music4dance-help/subscriptions/"
                  >purchasing a premium</a
                >
                subscriptions. This is one of the ways that you can help
                <a href="/home/contribute">support the music4dance project</a>
                (and you never have to look at ads on the site again - well, at least as long as
                your subscription lasts).
              </p>
            </BCardText>
          </BCardBody>
        </BCollapse>
      </BCard>

      <BCard no-body class="mb-1">
        <BCardHeader header-tag="header" class="p-1" role="tab">
          <BButton v-b-toggle.accordion-3b block variant="primary"
            >Why should I purchase a subscription to music4dance.net?</BButton
          >
        </BCardHeader>
        <BCollapse id="accordion-3b" accordion="my-accordion" role="tabpanel">
          <BCardBody>
            <BCardText>
              <p>
                <a href="https://music4dance.blog/music4dance-help/subscriptions/"
                  >Purchasing a premium subscription</a
                >
                is one of the ways that you can help
                <a href="/home/contribute">support the music4dance project.</a> It currently costs
                us more to keep the website running than we collect in advertising, referal and
                subscription fees. So if you enjoy using this site, please consider purchasing a
                <a href="/home/contribute">subscription or donating</a>.
              </p>
              <p>
                There are some additional
                <a href="https://music4dance.blog/music4dance-help/subscriptions/">perks</a> that
                come with a subscription:
              </p>
              <ul>
                <li>Advertisements are removed.</li>
                <li>
                  You will have access to Access to
                  <a href="https://music4dance.blog/music4dance-help/bonus-content/"
                    >bonus content</a
                  >
                  – songs that we weren’t able to match to a publisher or songs that have not yet
                  been tagged with a dance style.
                </li>
                <li>
                  You can
                  <a href="https://music4dance.blog/2019/09/22/create-a-spotify-playlist/"
                    >create a Spotify playlists</a
                  >
                  from search results.
                </li>
              </ul>
            </BCardText>
          </BCardBody>
        </BCollapse>
      </BCard>

      <BCard no-body class="mb-1">
        <BCardHeader header-tag="header" class="p-1" role="tab">
          <BButton v-b-toggle.accordion-4 block variant="primary">
            I can't find the song I'm looking for, what should I do?
          </BButton>
        </BCardHeader>
        <BCollapse id="accordion-4" accordion="my-accordion" role="tabpanel">
          <BCardBody>
            <BCardText>
              <p>
                You can go ahead and
                <a href="https://music4dance.blog/music4dance-help/add-songs/">add it</a>
                yourself.
              </p>
              <p>
                If you you're looking for the song because you don't know what to dance to it,
                please contact us at
                <a href="mailto:info@music4dance.net">info@music4dance.net</a>
                or through our
                <a href="https://music4dance.blog/feedback/">feedback form</a>
                and we will add it if we can.
              </p>
            </BCardText>
          </BCardBody>
        </BCollapse>
      </BCard>

      <BCard no-body class="mb-1">
        <BCardHeader header-tag="header" class="p-1" role="tab">
          <BButton v-b-toggle.accordion-5 block variant="primary">
            I'm a competition Ballroom dancer, how can music4dance help me find music?
          </BButton>
        </BCardHeader>
        <BCollapse id="accordion-5" accordion="my-accordion" role="tabpanel">
          <BCardBody>
            <BCardText>
              <p>
                Go to our
                <a href="/dances/ballroom-competition-categories"
                  >Ballroom competition categories</a
                >
                page and see the list of the different competition categories and the dance/song
                requirements for several organizations that publish competition rules.
              </p>
              <p>
                If you would like more information check out our related blog post.
                <a
                  href="https://music4dance.blog/im-a-ballroom-dancer-can-i-find-practice-songs-that-are-at-competition-tempo-revisited/"
                >
                  I'm a competition ballroom dancer, can I find practice songs that are a specific
                  tempo?</a
                >
              </p>
            </BCardText>
          </BCardBody>
        </BCollapse>
      </BCard>

      <BCard no-body class="mb-1">
        <BCardHeader header-tag="header" class="p-1" role="tab">
          <BButton v-b-toggle.accordion-6 block variant="primary">
            I want to find a great song for my first dance at my wedding, where should I look?
          </BButton>
        </BCardHeader>
        <BCollapse id="accordion-6" accordion="my-accordion" role="tabpanel">
          <BCardBody>
            <BCardText>
              <p>
                Check out the
                <a href="/dances/wedding-music">wedding page</a> for information to help you get
                started, or you can go to the
                <a href="/song/advancedsearchform">advanced search</a> page and look for songs that
                are tagged as First Dance and/or Wedding.
              </p>
              <p>
                If you would like more information check out our related blog posts.
                <a
                  href="https://music4dance.blog/wed-like-to-dance-a-real-partner-dance-as-the-first-dance-at-our-wedding-part-i-we-already-chose-our-song/"
                >
                  Wedding Music Part I: Can we dance the Foxtrot to our song?</a
                >
                and
                <a
                  href="https://music4dance.blog/wed-like-to-dance-a-real-partner-dance-as-the-first-dance-at-our-wedding-part-ii-we-already-chose-our-dance/"
                >
                  Wedding Music Part II: We’re learning to Rumba, help us find a good song for our
                  first dance</a
                >
              </p>
            </BCardText>
          </BCardBody>
        </BCollapse>
      </BCard>

      <BCard no-body class="mb-1">
        <BCardHeader header-tag="header" class="p-1" role="tab">
          <BButton v-b-toggle.accordion-7 block variant="primary">
            Not finding the answer to your question or want to give us feedback?
          </BButton>
        </BCardHeader>
        <BCollapse id="accordion-7" accordion="my-accordion" role="tabpanel">
          <BCardBody>
            <BCardText>
              <p>
                If you are not finding the answer to your question here or wish to give us feedback
                on any part of the site please contact us at
                <a href="mailto:info@music4dance.net">info@music4dance.net</a>
                or through our
                <a href="https://music4dance.blog/feedback/">feedback form</a>.
              </p>
            </BCardText>
          </BCardBody>
        </BCollapse>
      </BCard>
    </div>
  </PageFrame>
</template>
