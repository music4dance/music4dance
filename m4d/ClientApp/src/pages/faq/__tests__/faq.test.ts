import { describe, test } from "vitest";
import { testPageSnapshot } from "@/helpers/TestHelpers";
import App from "../App.vue";

describe("Faq", () => {
  test("Renders Faq Page", () => {
    testPageSnapshot(App);
  });
});
