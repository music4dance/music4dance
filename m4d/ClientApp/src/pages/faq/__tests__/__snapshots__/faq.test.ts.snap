// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Faq > Renders Faq Page 1`] = `
"<div><span context=\\"[object Object]\\">MainMenu</span>
  <nav aria-label=\\"breadcrumb\\">
    <nav aria-label=\\"breadcrumb\\" style=\\"padding: 0.25rem 0.5rem;\\">
      <ol class=\\"breadcrumb\\">
        <li class=\\"breadcrumb-item\\"><a class=\\"\\" href=\\"/\\">music4dance</a></li>
        <li class=\\"breadcrumb-item\\"><a class=\\"\\" href=\\"/home/info\\">Info</a></li>
        <li class=\\"breadcrumb-item active\\"><span aria-current=\\"location\\">FAQ</span></li>
      </ol>
    </nav>
  </nav>
  <div id=\\"body-content\\" class=\\"container-fluid body-content\\">
    <h1>Frequently Asked Questions</h1>
    <div>
      <div role=\\"tablist\\">
        <div class=\\"card mb-1\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"card-header p-1\\" header-tag=\\"header\\" role=\\"tab\\"><button class=\\"btn btn-md btn-primary collapsed\\" type=\\"button\\" block=\\"\\" aria-controls=\\"accordion-1\\" aria-expanded=\\"false\\"> Where can I find songs that work with the dance style I'm currently practicing? </button></div>
          <div id=\\"accordion-1\\" class=\\"collapse\\" is-nav=\\"false\\" accordion=\\"my-accordion\\" role=\\"tabpanel\\">
            <div class=\\"card-body\\">
              <!---->
              <!---->
              <p class=\\"card-text\\">
              <p> From the \\"Music Menu\\" at the top of the page, choose <a href=\\"/song\\">Song Library</a>, then click on the \\"All Dances\\" button in the upper left to choose the style of dance you are interestaed in. Or from the same menu choose <a href=\\"/song/advancedsearchform\\">Advanced Search</a> and include the style of dance you are interested then click search. In either case, the result should be a list of songs that can be danced to that style. </p>
              <p> If you would like more information check out our related blog posts. <a href=\\"https://music4dance.blog/question-1-im-learning-to-cha-cha-where-is-some-great-music-for-practicing/\\"> I’m learning to Cha Cha, where is some great music for practicing?</a> and <a href=\\"https://music4dance.blog/i-am-learning-the-foxtrot-where-can-i-find-some-music/\\"> I am learning the Foxtrot, where can I find some music?</a></p>
              <p> The <a href=\\"https://music4dance.blog/music4dance-help/advanced-search/\\">Advanced Search Documentation</a> describes the different search attributes in more detail. </p>
              </p>
            </div>
          </div>
          <!---->
          <!---->
        </div>
        <div class=\\"card mb-1\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"card-header p-1\\" header-tag=\\"header\\" role=\\"tab\\"><button class=\\"btn btn-md btn-primary collapsed\\" type=\\"button\\" block=\\"\\" aria-controls=\\"accordion-2\\" aria-expanded=\\"false\\"> What dance styles work with the song I'm listening to or a specific song that I know of? </button></div>
          <div id=\\"accordion-2\\" class=\\"collapse\\" is-nav=\\"false\\" accordion=\\"my-accordion\\" role=\\"tabpanel\\">
            <div class=\\"card-body\\">
              <!---->
              <!---->
              <p class=\\"card-text\\">
              <p> With the <a href=\\"/home/counter\\">tempo counter tool</a> you can tap out the beat while listening to a song, and we'll show you a list of styles that can be danced to the beat you just tapped out as well as links to the dance styles' information pages. </p>
              <p> Or if you have a particular song in mind, <a href=\\"/song/advancedsearchform\\">search</a> for it and if it is in the database look at what styles and tags are associated with it. Better yet, if you're listening to it in Spotify or Apple Music, you can drag the song to the text control on the <a href=\\"/song\\">song library page</a> or <a href=\\"/song/advancedsearchform\\">advanced search page</a> and if it's in the music4dance catalog, we'll take you right to it. See <a href=\\"https://music4dance.blog/new-feature-searching-for-a-song-from-spotify-or-itunes/\\">Searching for a song from Spotify or iTunes</a> for more details </p>
              <p> If you would like more information please read this blog blost <a href=\\"https://music4dance.blog/question-2-what-dance-styles-can-i-dance-to-my-favorite-songs/\\">What dance styles can I dance to my favorite song(s)?</a> and the <a href=\\"https://music4dance.blog/music4dance-help/tempo-counter/\\">documentation</a> for the Tempo Counter. </p>
              </p>
            </div>
          </div>
          <!---->
          <!---->
        </div>
        <div class=\\"card mb-1\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"card-header p-1\\" header-tag=\\"header\\" role=\\"tab\\"><button class=\\"btn btn-md btn-primary collapsed\\" type=\\"button\\" block=\\"\\" aria-controls=\\"accordion-3\\" aria-expanded=\\"false\\">Why should I sign up with Music4Dance?</button></div>
          <div id=\\"accordion-3\\" class=\\"collapse\\" is-nav=\\"false\\" accordion=\\"my-accordion\\" role=\\"tabpanel\\">
            <div class=\\"card-body\\">
              <!---->
              <!---->
              <p class=\\"card-text\\">
              <p><b>It's Free!</b> You'll be able to edit tags on songs and Music4Dance will remember the songs you like, edited, or even disliked. You'll alo be able to <a href=\\"https://music4dance.blog/music4dance-help/tag-editing/\\">tag songs</a>, <a href=\\"https://music4dance.blog/music4dance-help/dance-tags/\\">vote on the dancability of songs</a> and <a href=\\"https://music4dance.blog/music4dance-help/add-songs/\\">add your own songs</a>. <a href=\\"/identity/account/register\\">Register today!</a> We are always working on new features for our site and special features for our registered users. </p>
              <p> Once you've signed up, please consider <a href=\\"https://music4dance.blog/music4dance-help/subscriptions/\\">purchasing a premium</a> subscriptions. This is one of the ways that you can help <a href=\\"/home/contribute\\">support the music4dance project</a> (and you never have to look at ads on the site again - well, at least as long as your subscription lasts). </p>
              </p>
            </div>
          </div>
          <!---->
          <!---->
        </div>
        <div class=\\"card mb-1\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"card-header p-1\\" header-tag=\\"header\\" role=\\"tab\\"><button class=\\"btn btn-md btn-primary collapsed\\" type=\\"button\\" block=\\"\\" aria-controls=\\"accordion-3b\\" aria-expanded=\\"false\\">Why should I purchase a subscription to music4dance.net?</button></div>
          <div id=\\"accordion-3b\\" class=\\"collapse\\" is-nav=\\"false\\" accordion=\\"my-accordion\\" role=\\"tabpanel\\">
            <div class=\\"card-body\\">
              <!---->
              <!---->
              <p class=\\"card-text\\">
              <p><a href=\\"https://music4dance.blog/music4dance-help/subscriptions/\\">Purchasing a premium subscription</a> is one of the ways that you can help <a href=\\"/home/contribute\\">support the music4dance project.</a> It currently costs us more to keep the website running than we collect in advertising, referal and subscription fees. So if you enjoy using this site, please consider purchasing a <a href=\\"/home/contribute\\">subscription or donating</a>. </p>
              <p> There are some additional <a href=\\"https://music4dance.blog/music4dance-help/subscriptions/\\">perks</a> that come with a subscription: </p>
              <ul>
                <li>Advertisements are removed.</li>
                <li> You will have access to Access to <a href=\\"https://music4dance.blog/music4dance-help/bonus-content/\\">bonus content</a> – songs that we weren’t able to match to a publisher or songs that have not yet been tagged with a dance style. </li>
                <li> You can <a href=\\"https://music4dance.blog/2019/09/22/create-a-spotify-playlist/\\">create a Spotify playlists</a> from search results. </li>
              </ul>
              </p>
            </div>
          </div>
          <!---->
          <!---->
        </div>
        <div class=\\"card mb-1\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"card-header p-1\\" header-tag=\\"header\\" role=\\"tab\\"><button class=\\"btn btn-md btn-primary collapsed\\" type=\\"button\\" block=\\"\\" aria-controls=\\"accordion-4\\" aria-expanded=\\"false\\"> I can't find the song I'm looking for, what should I do? </button></div>
          <div id=\\"accordion-4\\" class=\\"collapse\\" is-nav=\\"false\\" accordion=\\"my-accordion\\" role=\\"tabpanel\\">
            <div class=\\"card-body\\">
              <!---->
              <!---->
              <p class=\\"card-text\\">
              <p> You can go ahead and <a href=\\"https://music4dance.blog/music4dance-help/add-songs/\\">add it</a> yourself. </p>
              <p> If you you're looking for the song because you don't know what to dance to it, please ontact us at <a href=\\"mailto:info@music4dance.net\\">info@music4dance.net</a> or through our <a href=\\"https://music4dance.blog/feedback/\\">feedback form</a> and we will add it if we can. </p>
              </p>
            </div>
          </div>
          <!---->
          <!---->
        </div>
        <div class=\\"card mb-1\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"card-header p-1\\" header-tag=\\"header\\" role=\\"tab\\"><button class=\\"btn btn-md btn-primary collapsed\\" type=\\"button\\" block=\\"\\" aria-controls=\\"accordion-5\\" aria-expanded=\\"false\\"> I'm a competition Ballroom dancer, how can music4dance help me find music? </button></div>
          <div id=\\"accordion-5\\" class=\\"collapse\\" is-nav=\\"false\\" accordion=\\"my-accordion\\" role=\\"tabpanel\\">
            <div class=\\"card-body\\">
              <!---->
              <!---->
              <p class=\\"card-text\\">
              <p> Go to our <a href=\\"/dances/ballroom-competition-categories\\">Ballroom competition categories</a> page and see the list of the different competition categories and the dance/song requirements for several organizations that publish competition rules. </p>
              <p> If you would like more information check out our related blog post. <a href=\\"https://music4dance.blog/im-a-competition-ballroom-dancer-can-i-find-practice-songs-that-are-a-specific-tempo/\\"> I'm a competition ballroom dancer, can I find practice songs that are a specific tempo?</a></p>
              </p>
            </div>
          </div>
          <!---->
          <!---->
        </div>
        <div class=\\"card mb-1\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"card-header p-1\\" header-tag=\\"header\\" role=\\"tab\\"><button class=\\"btn btn-md btn-primary collapsed\\" type=\\"button\\" block=\\"\\" aria-controls=\\"accordion-6\\" aria-expanded=\\"false\\"> I want to find a great song for my first dance at my wedding, where should I look? </button></div>
          <div id=\\"accordion-6\\" class=\\"collapse\\" is-nav=\\"false\\" accordion=\\"my-accordion\\" role=\\"tabpanel\\">
            <div class=\\"card-body\\">
              <!---->
              <!---->
              <p class=\\"card-text\\">
              <p> Check out the <a href=\\"/dances/wedding-music\\">wedding page</a> for information to help you get started, or you can go to the <a href=\\"/song/advancedsearchform\\">advanced search</a> page and look for songs that are tagged as First Dance and/or Wedding. </p>
              <p> If you would like more information check out our related blog posts. <a href=\\"https://music4dance.blog/wedding-music-part-i-can-we-dance-the-foxtrot-to-our-song/\\"> Wedding Music Part I: Can we dance the Foxtrot to our song?</a> and <a href=\\"https://music4dance.blog/wedding-music-part-ii-were-learning-to-rumba-help-us-find-a-good-song-for-our-first-dance/\\"> Wedding Music Part II: We’re learning to Rumba, help us find a good song for our first dance</a></p>
              </p>
            </div>
          </div>
          <!---->
          <!---->
        </div>
        <div class=\\"card mb-1\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"card-header p-1\\" header-tag=\\"header\\" role=\\"tab\\"><button class=\\"btn btn-md btn-primary collapsed\\" type=\\"button\\" block=\\"\\" aria-controls=\\"accordion-update\\" aria-expanded=\\"false\\"> Why are some pages styled differently than others? </button></div>
          <div id=\\"accordion-update\\" class=\\"collapse\\" is-nav=\\"false\\" accordion=\\"my-accordion\\" role=\\"tabpanel\\">
            <div class=\\"card-body\\">
              <!---->
              <!---->
              <p class=\\"card-text\\">
              <p> We're currently migrating from vue.js 2.0 to vue.js 3.0 and <a href=\\"https://bootstrap-vue.org/\\" target=\\"_blank\\">bootstrap-vue</a> to <a href=\\"https://bootstrap-vue-next.github.io/bootstrap-vue-next/\\" target=\\"_blank\\">bootstrap-vue-next</a>. The new pages should look very close to the old ones, but can be readily identified by the fact that the drop-down menus are rendered white text on a black background. We would appreciate your help as we roll the changes out. Please <a href=\\"https://music4dance.blog/bug-report/\\" target=\\"_blank\\">report any bugs you find</a> and we'll do our best to keep the site stable. </p>
              <p> This kind of update is occassionaly necessary to keep the site running. In this case, vue.js 2.0 is <a href=\\"https://v2.vuejs.org/eol/\\">no longer supported</a>. So we're forced to do a significant rewrite of the site to upgrade to Vue.js 3.0. This (and other similar updates) are quite time-consuming, so please consider <a href=\\"/home/contribute\\">subscribing or donating</a> to support these efforts. </p>
              </p>
            </div>
          </div>
          <!---->
          <!---->
        </div>
        <div class=\\"card mb-1\\">
          <!---->
          <!---->
          <!---->
          <div class=\\"card-header p-1\\" header-tag=\\"header\\" role=\\"tab\\"><button class=\\"btn btn-md btn-primary collapsed\\" type=\\"button\\" block=\\"\\" aria-controls=\\"accordion-7\\" aria-expanded=\\"false\\"> Not finding the answer to your question or want to give us feedback? </button></div>
          <div id=\\"accordion-7\\" class=\\"collapse\\" is-nav=\\"false\\" accordion=\\"my-accordion\\" role=\\"tabpanel\\">
            <div class=\\"card-body\\">
              <!---->
              <!---->
              <p class=\\"card-text\\">
              <p> If you are not finding the answer to your question here or wish to give us feedback on any part of the site please contact us at <a href=\\"mailto:info@music4dance.net\\">info@music4dance.net</a> or through our <a href=\\"https://music4dance.blog/feedback/\\">feedback form</a>. </p>
              </p>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </div>
  <div id=\\"footer-content\\">
    <hr>
    <footer>
      <p> © 2024 - <a href=\\"https://www.music4dance.net\\">Music4Dance.net</a> - <a href=\\"https://www.music4dance.net/home/sitemap\\">Site Map</a> - <a href=\\"https://www.music4dance.net/home/termsofservice\\">Terms of Service</a> - <a href=\\"https://www.music4dance.net/home/privacypolicy\\">Privacy Policy</a> - <a href=\\"https://www.music4dance.net/home/credits\\">Credits</a> - <a>Help</a></p>
    </footer>
  </div>
</div>"
`;
